readthedocs/projects/models.py

```diff
@@ -123,6 +123,60 @@ def subproject_prefix(self):
         return unsafe_join_url_path(prefix, self.alias, "/")
 
 
+class ProjectAddonsConfig(TimeStampedModel):
+
+    """
+    Addons project configuration.
+
+    Store all the configuration for each of the addons.
+    Everything is enabled by default.
+    """
+
+    DOC_DIFF_DEFAULT_ROOT_SELECTOR = "[role=main]"
+
+    # Analytics
+    analytics_enabled = models.BooleanField(default=True)
+
+    # Docdiff
+    doc_diff_enabled = models.BooleanField(default=True)
+    doc_diff_root_selector = models.CharField(null=True, blank=True)
+    doc_diff_show_additions = models.BooleanField(default=True)
+    doc_diff_show_deletions = models.BooleanField(default=True)
+    doc_diff_root_selector = models.CharField(null=True, blank=True)
+
+    # External version warning
+    external_version_warning = models.BooleanField(default=True)
+
+    # EthicalAds
+    ethicalads_enabled = models.BooleanField(default=True)
+
+    # Flyout
+    flyout_enabled = models.BooleanField(default=True)
+
+    # Hotkeys
+    hotkeys_enabled = models.BooleanField(default=True)
+
+    # Search
+    search_enabled = models.BooleanField(default=True)
+    search_default_filter = models.CharField(null=True, blank=True)
+
+    # Stable/Latest version warning
+    stable_latest_version_warning = models.BooleanField(default=True)
+
+
+class AddonSearchFilter(TimeStampedModel):
+
+    """
+    Addon search user defined filter.
+
+    Specific filter defined by the user to show on the search modal.
```

Comment:
An example or use case here would probably be useful. Is this to allow users to default to subprojects enabled or similar?

Reply:
The idea behind this is allow users to create filters to show in the search modal, like these ones:

![Screenshot_2023-09-14_10-26-27](https://github.com/readthedocs/readthedocs.org/assets/244656/18940734-f353-4194-8fff-2f878655d742)

Reply:
We can skip it for now if we are not sure how we are going to use this, tho.

Reply:
I don't feel too strongly, just wanted to make sure we had a plan for this. I'm fine to include it if we can make the docstring obvious for the user, but perhaps it's better for the next version? 🤷

Reply:
Making these filters configurable has been the idea from the beginning of the re-work for the search as you type. However, they are not yet configurable by the user and they won't be in this iteration because it will require more UI.

For now, we are only allowing them to opt-in into _all the addons at once_. In the following iteration we will be exposing specific configuration for each addon and they will be able to create these filters. This PR only creates the models we will use in the future.